nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteStreamCommand.java

```diff
@@ -342,16 +393,45 @@ public void onTrigger(ProcessContext context, final ProcessSession session) thro
                 final InputStream pis = process.getInputStream();
                 final InputStream pes = process.getErrorStream();
                 final BufferedInputStream bis = new BufferedInputStream(pis);
+                final BufferedInputStream bes = new BufferedInputStream(pes);
                 final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(pes))) {
             int exitCode = -1;
             final BufferedOutputStream bos = new BufferedOutputStream(pos);
+            final StringBuilder strBldr = new StringBuilder();
             FlowFile outputFlowFile = putToAttribute ? inputFlowFile : session.create(inputFlowFile);
+            FlowFile errorFlowFile=null;
+            FlowWriterThread flowWriter = null;
+            if (REDIRECT_ERROR_ERROR_STREAM.equals(redirectErrorStream)) {
+                 errorFlowFile = session.create(inputFlowFile);
+                flowWriter = new FlowWriterThread(errorFlowFile, session, bes);
+                this.executor.submit(flowWriter);
+            }
+            //if redirect error stream is set to log, then write the error to nifi component log.
+            if (REDIRECT_ERROR_LOG.equals(redirectErrorStream)) {
+                this.executor.submit(new Runnable() {
+                    @Override
+                    public void run() {
+                       try (final BufferedReader reader = new BufferedReader(new InputStreamReader(pes))) {
+                            String line;
+                            while ((line = bufferedReader.readLine()) != null) {
+                                 logger.warn("ExecuteStreamCommand :"+line);
+                                 strBldr.append(line).append("\n");
```

Comment:
I think this probably warrants its own jira as it was existing behavior, I'm going to write it up separately